pubsub/gcppubsub/gcppubsub.go

```diff
@@ -80,8 +80,9 @@ import (
 var endPoint = "pubsub.googleapis.com:443"
 
 var sendBatcherOpts = &batcher.Options{
-	MaxBatchSize: 1000, // The PubSub service limits the number of messages in a single Publish RPC
-	MaxHandlers:  2,
+	MaxBatchSize:     1000, // The PubSub service limits the number of messages in a single Publish RPC
+	MaxHandlers:      2,
+	MaxBatchByteSize: 10 * 1000 * 1000, // The PubSub service limits the size of request body in a single Publish RPC
```

Comment:
MB is presumably 1024 * 1024.

Reply:
Be ready for more confusion: https://en.wikipedia.org/wiki/Byte#Multiple-byte_units

Reply:
That's what I did initially and actually hit and other request size validation error on the pubsub API when using 1024*1024.
I feel there might be different units used in the pubsub API itself (maybe at loadbalancer level vs grpc handlers ?)

Reply:
I suspect they are enforcing the total bytes for the request, while this only measures the sum of the Body fields for each message (ignores Attributes plus all other metadata). It is not obvious from the GCP documentation though.

How about we use 9 * 1024 * 1024 as the limit and document that the actual limit is 10 * 1024 * 1024 but we're giving ourselves some headroom for metadata?